src/providers/mbed_provider/key_management.rs

```diff
@@ -43,13 +48,12 @@ fn create_key_id(
     key_attributes: Attributes,
     store_handle: &mut dyn ManageKeyInfo,
     local_ids_handle: &mut LocalIdStore,
-) -> Result<psa_key_id_t> {
-    let mut key_id = rand::random::<psa_key_id_t>();
-    while local_ids_handle.contains(&key_id)
-        || key_id == 0
-        || key_id > PSA_MAX_PERSISTENT_KEY_IDENTIFIER
-    {
-        key_id = rand::random::<psa_key_id_t>();
+) -> Result<key::psa_key_id_t> {
+    let mut rng = SmallRng::from_entropy();
+    let mut key_id = rng.gen_range(key::PSA_KEY_ID_USER_MIN, key::PSA_KEY_ID_USER_MAX);
```

Comment:
`gen_range` might be exclusive of its `high` [parameter](https://docs.rs/rand/0.7.3/rand/trait.Rng.html#method.gen_range) whereas (to be confirmed) `PSA_KEY_ID_USER_MAX` is inclusive. It might then need to be `key::PSA_KEY_ID_USER_MAX + 1` for the second parameter.

Reply:
ah yes, my bad. I'd expect `PSA_KEY_ID_USER_MAX` to be inclusive.